elf_reader_test.go

```diff
@@ -543,24 +507,10 @@ func TestTailCall(t *testing.T) {
 			t.Fatal(err)
 		}
 
-		if spec.ByteOrder != internal.NativeEndian {
-			return
-		}
-
-		var obj struct {
-			TailMain  *Program `ebpf:"tail_main"`
-			ProgArray *Map     `ebpf:"prog_array_init"`
-		}
-
-		err = spec.LoadAndAssign(&obj, nil)
```

Comment:
The rationale for using L&A here was testing if prog array lazy loading works correctly. Any chance we can keep this test intact? Might need a comment to clarify this so we don't accidentally remove it later.

I think having a `newCollection` that uses L&A would be useful here. Then we get the CollectionSpec equality checks, byteorder and ErrNotSupported skips out of the box for these tests too.

Reply:
Isn't that covered by TestUnassignedProgArray?

Reply:
That just tests the error case, doesn't do an actual prog_run to make sure the tail calls are hooked up properly.